### PR TITLE
chore: test changeset experimental flag

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-2323]

### Description

Setup changeset config to update peer dependencies only when out of range. And also update ranges in peer dependencies so we are in control to update peer dependencies.

**The problem:**
Every `minor` update to `@toptal/picasso` triggers `major` bump for `@toptal/picasso-forms`, `@toptal/picasso-charts`, and analytics for charts. The reason behind it is, that these packages are having Picasso in peer dependencies, when peer dependency is updated, it is considered as a possible breaking change. To solve this issue we will upgrade peer dependencies only for major bumps.

### How to test

- check the screenshots

### screenshots
**`minor` bump of Picasso with previous setup**
image

**`minor` bump of Picasso with new config**
image

**`major` bump of Picasso with new config**
image

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and
1. You have no extra requests.
2. You have optional requests.
   1. Add `nit: ` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because
1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:
1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2323]: https://toptal-core.atlassian.net/browse/FX-2323?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/stale-lamps-walk.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso': minor
```
Comment (TomasSlama) [inline .changeset/stale-lamps-walk.md:2]: this file needs to be removed before merging.

File: packages/picasso-charts/package.json [hunk 2]
```diff
@@ -24,8 +24,8 @@
   "homepage": "https://github.com/toptal/picasso/tree/master/packages/picasso-charts#readme",
   "peerDependencies": {
     "@material-ui/core": "4.11.0",
-    "@toptal/picasso": "^17.0.0",
-    "@toptal/picasso-shared": "^7.1.1",
+    "@toptal/picasso": ">=17.0.0",
+    "@toptal/picasso-shared": ">=7.1.1",
```
Comment (dmaklygin1) [inline packages/picasso-charts/package.json:28]: I'm thinking about these peer dependencies with sign: `>=`, seems it allows also 17, 18, 19, etc versions of `@toptal/picasso` 
won't this vesion be upgraded after the new changeset is generated?

File: .changeset/config.json [hunk 3]
```diff
@@ -6,5 +6,9 @@
   "access": "public",
   "baseBranch": "master",
   "updateInternalDependencies": "patch",
-  "ignore": []
+
+  "ignore": [],
+  "___experimentalUnsafeOptions_WILL_CHANGE_IN_PATCH": {
+    "onlyUpdatePeerDependentsWhenOutOfRange": true
+  }
```
Comment (dmaklygin1) [inline .changeset/config.json:13]: I was thinking....According to [task](https://toptal-core.atlassian.net/browse/FX-2323), the real problem was about bumping major versions of internal packages when there is `patch`.
does it really solve the problem, could you please explain a little bit :D

File: packages/picasso-charts/package.json (hunk 2 above)
Comment (TomasSlama) [reply packages/picasso-charts/package.json:28]: I will try it out, but I hope it will not.

File: .changeset/config.json (hunk 3 above)
Comment (TomasSlama) [reply .changeset/config.json:13]: the problem was, when picasso is bumbed with `minor` version. It triggered `major` bumbs in `picasso-lab`, `picasso-charts`, `picasso-forms` and `analytics-charts`.

Without this flag peer dependency would be upgraded and major bumbs would appear.

File: .changeset/blue-kids-reply.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso': minor
+---
+
+test
```
Comment (TomasSlama) [inline .changeset/blue-kids-reply.md:5]: this file will needs to be removed after reviews.

File: packages/picasso-charts/package.json (hunk 2 above)
Comment (TomasSlama) [reply packages/picasso-charts/package.json]: I decided to remove `>=`, we would have to upgrade peer dependencies manually in the future, when Picasso's major change would affect some of the other libraries, and I think we would forget about it. This way we will update only with major bumps of Picasso and I would say it is ok. Definitely better than the current solution.

**Review by dmaklygin — APPROVED** [review]

**Review by separatio — APPROVED** [review]
Changes look ok. I have no idea how to test this. Screenshots look :ok:

**Review by ozgurkececioglu — APPROVED** [review]

**Pull request merged**